### PR TITLE
Socket (intro, frame)

### PR DESCRIPTION
**Pull request by evanyang1**

### CODE REVIEW

**Comment by bonnieli** [comment]
I'm guessing this PR is not ready to be reviewed as it's not working just yet - @evanyang1 next time please create a Draft PR rather than an actual PR if it's not ready

File: app.js [hunk 1]
```diff
@@ -13,6 +13,24 @@ const { json, urlencoded } = express;
...
+          "Access-Control-Allow-Credentials": true
+      };
+      res.writeHead(200, headers);
+      res.end();
+  },
+  cors: {
+    origin: "https://example.com",
+    methods: ["GET", "POST"],
+    allowedHeaders: ["my-custom-header"],
+    credentials: true
```
Comment (ThinkTink) [inline app.js]: It's ok to start with examples from the internet, but we always need to adapt them to our use case - example.com and ['my-custom-header'] are just boilerplate information, for example.
Do we need both the handlePreflightRequest and cors? It's best to always test out how much code you can eliminate and go with the working solution that has the least amount of code - that keeps things cleaner and easier to maintain

File: client/src/components/Conversation.js [hunk 2]
```diff
@@ -18,6 +22,17 @@ const useStyles = makeStyles((theme) => ({
 
 const Conversation = (props) => {
   const classes = useStyles();
+
+  useEffect(() => {
+    let socket = io(serverURL);
```
Comment (ThinkTink) [inline client/src/components/Conversation.js]: const*

File: client/src/components/Navbar.js [hunk 3]
```diff
@@ -43,6 +43,12 @@ const useStyles = makeStyles((theme) => ({
 
 const Navbar = (props) => {
   const classes = useStyles();
+
+  const handleSwitch = () => {
+    // show original language
+    return
```
Comment (ThinkTink) [inline client/src/components/Navbar.js]: It's best to keep feature branches clean and only have code related to that feature so we don't accidentally merge in half finished features

**Review by ThinkTink — COMMENTED** [review]
This is a good start. Can you work on cleaning up what you have and adding some TODO comments to outline what next steps are needed?

**Pull request merged**